samza-api/src/main/java/org/apache/samza/system/IncomingMessageEnvelope.java

```diff
@@ -92,21 +92,55 @@ public IncomingMessageEnvelope(SystemStreamPartition systemStreamPartition, Stri
   }
 
   /**
-   * Getter for event time
+   * Getter for event time (Nanos)
    * @return this.eventTime
    */
-  public long getEventTime() {
+  public long getEventTimeNanos() {
     return eventTime;
   }
 
   /**
-   * Getter for arrival time
+   * Getter for arrival time (Nanos)
    * @return this.arrivalTime
    */
-  public long getArrivalTime() {
+  public long getArrivalTimeNanos() {
     return arrivalTime;
   }
 
+  /**
+   * For backward compatibility, Since arrivalTime was in Millis and getArrivalTime was returning
+   * arrivalTime (in Millis)
+   * @return arrivalTime in Millis
+   */
+  public long getArrivalTime() {
+    return getArrivalTimeMillis();
+  }
+
+  /**
+   * For backward compatibility, Since eventTime was in Millis and getEventTime was returning
+   * evenTime (in Millis)
+   * @return
+   */
+  public long getEventTime() {
+    return getEventTimeMillis();
+  }
+
+  /**
+   * Getter for event time in Millis
+   * @return this.eventTime / 10^6
+   */
+  public long getEventTimeMillis() {
+    return eventTime / MILLIS_IN_NANO;
+  }
+
+  /**
+   * Getter for arrival time in Millis
+   * @return this.arrivalTime / 10^6
+   */
+  public long getArrivalTimeMillis() {
```

Comment:
I'd like to understand what we're trying to achieve here. Backwards compatibility is fine, but now there are 6 time related methods in this class which is not helpful.

Reply:
essentially we need to capture latencies in nano or micro seconds resolution, because millis report zero. On local machine testing, 0's for queryLatency becomes non-zero when we use nano. So that is why we switched to nanoTime. To make this change backward compatible, we needed the old get and the new getNano... the getMillis are the natural counterparts for getNano, and the old get uses them...

Reply:
this is resolved now, we have only 1 new getter (3 total)